### DISCUSSION TITLE
Host multiple sites with single node instance

### DISCUSSION THREAD

**Discussion by jd-solanki** [nuxt/nuxt#17865] [Questions]
Hi,

We at [ThemeSelection](http://themeselection.com/) provide admin templates to bootstrap any project. If you noticed we deploy multiple demos of single template on our server: `/<template_name>/demo-<demo_number>/`. Before NuxtJS, we were using VueJS SPA and we just route all the routes of a particular demo via its index.html file. It was just like a static building serving.

Recently, we started working on NuxtJS and found that we can no longer follow our above described server configuration as nuxtjs requires serving the build differently.

The problem is:

## `static` build

As it generates a separate HTML file for each route with its own CSS and JS. It takes lots of space assuming we show details page of  400 table records. And if we look at the big picture, It generates the same HTML pages for multiple demos. If we even go a step upward we also have multiple templates (different designs) so static generation takes too much space on the server.

## `server` build

server build is simple just spin the node server and you are done. But as we have at least 6 demos of each template and assuming we have 4 templates. We need to run `6x4=24` node instances/servers for showcasing our demos. Which I don't think is appropriate.

---

So, I want to know is there any way, we can serve multiple demos (and multiple templates) with ease as we did with VueJS SPA.

Thanks.

**Comment by danielroe** [comment]
Is this meant to refer to Nuxt 3 or Nuxt 2?

**Comment by jd-solanki** [comment]
Sorry I didn't received any notification and lost it 🙏🏻
